File: frontend/src/api/project/project.ts
import request from '/@/utils/request';

/**
 * 项目管理接口
 */
export function useProjectApi() {
  return {
    // 获取项目列表
    getList: (data?: object) => {
      return request({
        url: '/project/list',
        method: 'POST',
        data,
      });
    },
    // 获取项目详情
    getById: (id: number) => {
      return request({
        url: '/project/get',
        method: 'POST',
        data: { id },
      });
    },
    // 新增项目
    create: (data?: object) => {
      return request({
        url: '/project/create',
        method: 'POST',
        data,
      });
    },
    // 更新项目
    update: (data?: object) => {
      return request({
        url: '/project/update',
        method: 'POST',
        data,
      });
    },
    // 删除项目
    delete: (id: number) => {
      return request({
        url: '/project/delete',
        method: 'POST',
        data: { id },
      });
    },
  };
}

File: backend/app/apis/system/project.py
from app.corelibs.custom_router import APIRouter
from app.utils.response import HttpResponse
from app.schemas.system.project import (
    ProjectIn, ProjectUpdate, ProjectDel, ProjectQuery,
)
from app.services.system.project import ProjectService

router = APIRouter()


@router.post('/create', description="新增项目信息")
async def create(params: ProjectIn):
    data = await ProjectService.create(params)
    return await HttpResponse.success(data)


@router.post('/update', description="更新项目信息")
async def update(params: ProjectUpdate):
    await ProjectService.update(params)
    return await HttpResponse.success()


@router.post('/delete', description="删除项目信息")
async def delete(params: ProjectDel):
    await ProjectService.delete(params)
    return await HttpResponse.success()


@router.post('/list', description="获取项目信息列表")
async def get_list(params: ProjectQuery):
    data = await ProjectService.get_list(params)
    return await HttpResponse.success(data)


@router.post('/get', description="根据ID获取项目信息")
async def get_by_id(params: ProjectDel):
    data = await ProjectService.get_by_id(params.id)
    return await HttpResponse.success(data)
